concepts/50 React Components/40 Component Configuration Syntax/55 Markup Customization/0 Markup Customization.md

```diff
@@ -1,5 +1,7 @@
-Templates allow you to customize UI components. In the DevExtreme API, template properties end with `Template`: `itemTemplate`, `groupTemplate`, `contentTemplate`. When you specify them in React, replace `Template` in the name with `Render` or `Component`, depending on whether the template is a rendering function or custom component. For example, instead of `itemTemplate`, use `itemRender` or `itemComponent`. If a component does not reinitiate a template when the component rerenders, you should use the `itemComponent` property. Otherwise, we recommend that you use the `itemRender` property. 
+You can define custom content in DevExtreme components. We supply 2 types of properties for this task: ones that end with *'render'* (for example, [cellRender](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellRender)), and ones that end with *'component'* (for example, [cellComponent](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellComponent)).
 
-If the property is called `template`, without any prefix, (in the [Button](/api-reference/10%20UI%20Components/dxButton/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxButton/Configuration/#template'), [Drawer](/api-reference/10%20UI%20Components/dxDrawer/1%20Configuration/template.md '/Documentation/ApiReference/UI_Components/dxDrawer/Configuration/#template'), and other UI components), use the `render` or `component` property instead.
+The *'render'* properties allow you to employ a rendering function. Whatever you return with this function is rendered in a virtual DOM. The rendering function is not a component: it lacks state and cannot utilize hooks. However, you can create the rendering function inside your component, enclosed within the [useCallback](https://react.dev/reference/react/useCallback). You can pass local variables as dependencies, allowing you to use them in your template. Refer to the following article to see an example: [Using a Rendering Function](/Documentation/Guide/React_Components/Component_Configuration_Syntax/#Markup_Customization/Using_a_Rendering_Function).
```

Comment:
We try not to use articles with plural nouns, unless we refer to a very specific group. So that's what I'm changing in the first sentence. 

In the second, I want to get rid of passive. I'm also not sure about the phrase "return with the function". So I'm turning this into an active phrase - "component calls... and renders ... ".

In the third, you refer to "the function". But I think you are actually stating the fact that is true for any function. You are not just talking about one specific function you mentioned earlier. So we need to return to indefinite article. 

"allowing you to use them" - that's the kind of ING word that's ambiguous. Maybe "dependencies" allow to do something, maybe "variables" allow to do something, maybe the fact that you "pass" variables allows you to do something. And it in the end, I don't even think the word "allowing" is necessary. 

```suggestion
*'Render'* properties allow you to employ rendering functions. A DevExpress control calls the specified function and renders the return value to the virtual DOM. Note that a rendering function is not a component: it lacks state and cannot utilize hooks. However, you can create a rendering function inside your component and enclose it within the [useCallback](https://react.dev/reference/react/useCallback) hook. You can pass local variables as dependencies and use them in your template. The following article contains an example: [Using a Rendering Function](/Documentation/Guide/React_Components/Component_Configuration_Syntax/#Markup_Customization/Using_a_Rendering_Function).
```